Neos.Flow/Classes/Cache/AnnotationsCacheFlusher.php

```diff
@@ -48,25 +52,40 @@ final class AnnotationsCacheFlusher
     protected $reflectionService;
 
     /**
-     * A slot that flushes caches as needed if classes with specific annotations have changed
+     * Caches to flush for a given annotation
+     *
+     * @var array in the format [<AnnotationClassName> => [<CacheName_1>, <CacheName_2>]]
+     */
+    private $annotationToCachesMap = [];
+
+    /**
+     * Register an annotation that should trigger a cache flush
+     *
+     * @param string $annotationClassName fully qualified class name of the annotation
+     * @param string[] $cacheNames Cache names to flush if a class containing the given annotation is compiled (e.g. ["Flow_Mvc_Routing_Route", Flow_Mvc_Routing_Resolve"])
+     */
+    public function registerAnnotation(string $annotationClassName, array $cacheNames): void
```

Comment:
Having this as instance method means you have to create the insance to add a new mapping. Not sure that's much of an issue, but it should at least be considered. The $annotationToCachesMap could as well be static.

Reply:
True. I don't think that this would be a real problem. And for the "core mappings" we only create the instance in the slot anyways

Reply:
Was mostly thinking 3rd-party now. Since otherwise the class would only be instanciated at compile time, user code would need to act "clever" to avoid instanciating it too eagerly.